feat(route): add HAFU 河南财政金融学院

deep1nlife opened the pull request:
<!-- 
如有疑问，请参考 https://github.com/DIYgod/RSSHub/discussions/8002
Reference: https://github.com/DIYgod/RSSHub/discussions/8002
-->

## 该 PR 相关 Issue / Involved issue

Close #10245 

## 完整路由地址 / Example for the proposed route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route

```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/hafu/news/ggtz
/hafu/news/jwc
/hafu/news/zsjyc
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [X] 新的路由 New Route
  - [X] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [X] 文档说明 Documentation
  - [X] 中文文档 CN
  - [ ] 英文文档 EN
- [X] 全文获取 fulltext
  - [X] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [X] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [X] 可以解析 Parsed
  - [X] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
@TonyRL Thanks for your patience in reviewing my previous crappy code on PR #10245 , now that I've rewritten them...  

Commit:
feat(route): add HAFU 河南财政金融学院

WenPeiTung commented:
> Successfully [generated](https://github.com/DIYgod/RSSHub/actions/runs/2740798384) as following:
> http://localhost:1200/hafu/news/:type/:limit? - Failed

The optional key word 'limit' that value is empty in the 'Route requested', that's normal? Do I need to fix it?

WenPeiTung commented:
> Successfully [generated](https://github.com/DIYgod/RSSHub/actions/runs/2740798384) as following:
> http://localhost:1200/hafu/news/:type/:limit? - Failed

I was confused last night, I updated the route example, please test it again.

TonyRL commented on lib/v2/hafu/router.js:
```diff
@@ -0,0 +1,3 @@
+module.exports = function (router) {
+    router.get('/news/:type/:limit?', require('./news'));
```
```suggestion
    router.get('/news/:type?', require('./news'));
```

TonyRL commented on lib/v2/hafu/utils.js:
```diff
@@ -0,0 +1,173 @@
+            const href = $(item).find('a[class=c269582]').attr('href').slice(3);
+            const link = typeMap.ggtz[1] + href;
+
+            const cache = await ctx.cache.get(link);
```
Please use ctx.cache.tryGet()
https://docs.rsshub.app/joinus/quick-start.html#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun

TonyRL commented on lib/v2/hafu/utils.js:
```diff
@@ -0,0 +1,173 @@
+                author,
+            };
+            //                                                 cache expiretime 120mins
+            ctx.cache.set(link, JSON.stringify(result), config.cache.contentExpire * 24);
```
Default value is enough

TonyRL commented on lib/v2/hafu/utils.js:
```diff
@@ -0,0 +1,173 @@
+
+            const author = header.substring(0, index - 2) || '';
+            // form articlePage || hostPage
+            const Date = header.substring(index + 3, index + 19) || $(item).find('a[class=c269582_date]');
```
```suggestion
            const date = header.substring(index + 3, index + 19) || $(item).find('a[class=c269582_date]').text();
```
camelCase
Missing `.text()`

TonyRL commented on lib/v2/hafu/utils.js:
```diff
@@ -0,0 +1,173 @@
+            const author = header.substring(0, index - 2) || '';
+            // form articlePage || hostPage
+            const Date = header.substring(index + 3, index + 19) || $(item).find('a[class=c269582_date]');
+            const pubDate = parseDate(Date, 'YYYY-MM-DD HH-mm') || parseDate(Date, 'YYYY-MM-DD  ');
```
```suggestion
            const pubDate = parseDate(Date, 'YYYY-MM-DD HH:mm') || parseDate(Date, 'YYYY-MM-DD');
```

TonyRL commented on lib/v2/hafu/templates/hafu.art:
```diff
@@ -0,0 +1,3 @@
+{{ if articleBody }}
+    {{@ articleBody }}
+{{ /if }}
```
I think user will be happier if you can add attachments

Commit:
fix problems

1. fix route path in all files
2. fix radar target()
3. use camelCase
4. use ctx.cache.tryGet()
5. add "attachments" in description

TonyRL commented on docs/university.md:
```diff
@@ -1252,6 +1252,18 @@ category 列表：
+
+### 河南财政金融学院
+
+\<Route author="WenPeiTung" example="/hafu/news/ggtz" path="/hafu/news/:type?" radar="1" rssbud="1" :paramsDesc="\[' 分类，见下表（默认为"ggtz")',' 支持全局参数"limit"，默认为"10"']">
```
```suggestion
<Route author="WenPeiTung" example="/hafu/news/ggtz" path="/hafu/news/:type?" radar="1" rssbud="1" :paramsDesc="['分类，见下表（默认为 `ggtz`)']">
```

TonyRL commented on lib/v2/hafu/utils.js:
```diff
@@ -0,0 +1,199 @@
+}
+// A. got from hostPage     1.article(link), 2.article(title), 3.(pubDate)
+// B. got from articlePage  1.description(fullText), 2.article(author), 3.detailed(pubDate)
+function ggtzParse(ctx, $) {
+    const data = $('a[class=c269582]').parent().slice(0, limit);
+    const resultItems = data
```
```suggestion
async function ggtzParse(ctx, $) {
    const data = $('a[class=c269582]').parent().slice(0, limit);
    const resultItems = await Promise.all(data.map(...));
    return resultItems;
}
```

TonyRL commented on lib/v2/hafu/utils.js:
```diff
@@ -0,0 +1,199 @@
+    const $ = cheerio.load(response.data);
+
+    limit = ctx.query.limit || limit;
+    const resultList = typeMap[type].parseFn(ctx, $);
```
```suggestion
    const resultList = await typeMap[type].parseFn(ctx, $);
```

Commit:
1.run format, 2.add Promise.all()

WenPeiTung commented:
finally...

WenPeiTung commented:
finally...
I added so much trouble for you, I feel sorry, I'll express my sincerity again, thank you！
